### PR TITLE
Reviews slice

### PR DESCRIPTION
**Pull request by JonathanHelvey**

### CODE REVIEW

File: client/store/reviews.js [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+const POST_REVIEW = 'POST_REVIEW'
+
+//ACTION CREATORS
+const getReviews = reviews => ({type: GET_ALL_REVIEWS, reviews})
+const postReview = review => ({type: POST_REVIEW, review})
+
+//THUNK CREATORS
+export const getAllReviews = () => async dispatch => {
+  try {
+    const res = await axios.get('/api/products/:id')
```
Comment (krnsk0) [inline client/store/reviews.js]: This line will need updating...

File: client/store/reviews.js [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+    dispatch(getReviews(res.data))
+  } catch (err) {
+    console.error(err)
+  }
+}
+
+export const writeReview = review => async dispatch => {
+  try {
+    const {data} = await axios.post(`/api/products/`, review)
```
Comment (krnsk0) [inline client/store/reviews.js]: Same here, I think?

File: client/store/reviews.js [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+// ACTION TYPES
+const GET_ALL_REVIEWS = 'GET_ALL_REVIEWS'
+const POST_REVIEW = 'POST_REVIEW'
+
+//ACTION CREATORS
+const getReviews = reviews => ({type: GET_ALL_REVIEWS, reviews})
+const postReview = review => ({type: POST_REVIEW, review})
+
+//THUNK CREATORS
+export const getAllReviews = () => async dispatch => {
```
Comment (krnsk0) [inline client/store/reviews.js]: It's worth thinking a bit about when we need to pull up reviews-- will we ever need to display *all* reviews? 

I can imagine us needing to display reviews by product, or perhaps by user... maybe best to start with displaying by product

File: server/api/products.js [hunk 4]
```diff
@@ -8,47 +8,76 @@ router.get('/page/:offset', async (req, res, next) => {
...
+
+//FIX GET REQUEST ABOVE
+router.post('/:id', async (req, res, next) => {
+  try {
+    console.log(req.body)
+    const review = await Reviews.create(req.body)
+    req.json(review)
   } catch (err) {
     next(err)
   }
```
Comment (krnsk0) [inline server/api/products.js]: I'm confused about how these routes are different from routes for posting/getting products

Comment (JonathanHelvey) [reply server/api/products.js]: Agree.  The routes I got confused.  I worked on them a bit tonight to put reviews in there own routes with product.  Need to review all of reviews slice Tomorrow with someone and go over routes.

**Pull request merged**